src/Microsoft.Diagnostics.Monitoring/DiagnosticServices.cs

```diff
@@ -0,0 +1,118 @@
+            NETCore.Client.DumpType dumpType = MapDumpType(mode);
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                // Get the process
+                Process process = Process.GetProcessById(pid);
+                await Dumper.CollectDumpAsync(process, dumpFilePath, dumpType);
+            }
+            else
+            {
+                await Task.Run(() =>
+                {
+                    var client = new DiagnosticsClient(pid);
+                    client.WriteDump(dumpType, dumpFilePath);
+                });
+            }
+
+            return new FileStreamWrapper(dumpFilePath);
+        }
+
+        public Task<Stream> StartCpuTrace(int pid, int durationSeconds)
+        {
+            if ((durationSeconds < 1) || (durationSeconds > MaxTraceSeconds))
+            {
+                throw new InvalidOperationException("Invalid duration");
+            }
+
+            //TODO Should we limit only 1 trace per file?
+            var client = new DiagnosticsClient(pid);
+
+            //TODO Pull event providers from the configuration.
+            var cpuProviders = new EventPipeProvider[] {
+                new EventPipeProvider("Microsoft-DotNETCore-SampleProfiler", System.Diagnostics.Tracing.EventLevel.Informational),
+                new EventPipeProvider("Microsoft-Windows-DotNETRuntime", System.Diagnostics.Tracing.EventLevel.Informational, (long)Tracing.Parsers.ClrTraceEventParser.Keywords.Default)
+            };
+
+            EventPipeSession session = client.StartEventPipeSession(cpuProviders, requestRundown: true);
+
+            CancellationToken token = _tokenSource.Token;
+            Task traceTask = Task.Run(async () =>
+            {
+                try
+                {
+                    await Task.Delay(TimeSpan.FromSeconds(durationSeconds), token);
+                }
+                finally
+                {
+                    session.Stop();
+                    //We rely on the caller to Dispose the EventStream file.
+                }
+            });
+
+            return Task.FromResult(session.EventStream);
+
+        }
+
+        private static NETCore.Client.DumpType MapDumpType(DumpType dumpType)
+        {
+            switch(dumpType)
+            {
+                case DumpType.Full:
+                    return NETCore.Client.DumpType.Full;
+                case DumpType.MiniWithHeap:
+                    return NETCore.Client.DumpType.WithHeap;
+                case DumpType.Triage:
+                    return NETCore.Client.DumpType.Triage;
+                case DumpType.Normal:
+                    return NETCore.Client.DumpType.Normal;
+                default:
+                    throw new ArgumentException("Unexpected dumpType", nameof(dumpType));
+            }
+        }
+
+        public void Dispose()
+        {
+            _tokenSource.Cancel();
+            _tokenSource.Dispose();
```

Comment:
Remove the dispose.  It could result it an exception if anyone else tries to access the token after this.